src/pages/SignupPage.jsx

```diff
@@ -0,0 +1,225 @@
+
+const SignupPage = () => {
+  const dispatch = useDispatch();
+  const { showToast } = useToast();
+  const navigate = useNavigate();
+  const location = useLocation();
+  let from = location.state?.from?.pathname ?? "/feed";
+
+  const [user, setUser] = useState({
+    firstName: "",
+    lastName: "",
+    username: "",
+    password: "",
+    confirmPassword: "",
+  });
+  const [passwordVisible, setPasswordVisible] = useState(false);
+  const [confirmPassVisible, setConfirmPassVisible] = useState(false);
+  const [signupError, setSignupError] = useState("");
+
+  const singupSubmitHandler = async (user) => {
+    console.log(user);
+    try {
+      const response = await dispatch(signupUser(user));
+      console.log("response from signup submit", response);
+      if (response?.error) {
+        throw new Error(error);
+      }
+      if (response?.payload.encodedToken) {
+        showToast("success", "Signed up and logged in successfully.");
+        navigate(from, { replace: true });
+      }
+    } catch (error) {
+      showToast("error", "Can't sign up. Recheck details and try again.");
+      setSignupError("Sign up failed.");
+    }
+  };
+  return (
+    <div className="container-fluid px-1 px-md-5 px-lg-1 px-xl-5 py-5 mx-auto">
+      <div className="card card0 border-0">
+        <div className="flex-row-centre">
+          <div className="col-lg-6 border-line">
+            <form
+              className="card2 card border-0 px-4 py-5"
+              onSubmit={(e) => {
+                e.preventDefault();
+                singupSubmitHandler(user);
+              }}
+            >
+              <div className="flex-col px-3">
+                <label htmlFor="firstname" className="mb-1 input-required">
+                  <p className="mb-1 text-sm d-inline">First Name</p>
+                </label>
+                <input
+                  id="firstname"
+                  className="mb-4 input-field p-1"
+                  type="text"
+                  name="firstname"
+                  placeholder="Enter your first name"
+                  required
+                  value={user.firstName}
+                  onChange={(e) =>
+                    setUser({ ...user, firstName: e.target.value })
+                  }
+                />
+              </div>
+              <div className="flex-col px-3">
+                <label htmlFor="lastname" className="mb-1 input-required">
+                  <p className="mb-1 text-sm d-inline">Last Name</p>
+                </label>
+                <input
+                  id="lastname"
+                  className="mb-4 input-field p-1"
+                  type="text"
+                  name="lastname"
+                  placeholder="Enter your last name"
+                  required
+                  value={user.lastName}
+                  onChange={(e) =>
+                    setUser({ ...user, lastName: e.target.value })
+                  }
```

Comment:
since there are many input fields to handle, you can write a changeHandler function and update state using key value pairs.